### DISCUSSION TITLE
Idea: Replace Nutrients System with Food Groups

### DISCUSSION THREAD

**Discussion by RobbieNeko** [cataclysmbn/Cataclysm-BN#3746] [General]
As inspired by a [Minecraft Mod](https://www.curseforge.com/minecraft/mc-mods/nutritional-balance), I propose replacing the current / past Nutrients System with more generalized Food Groups instead. This would help incentivize players to diversify their food intake without diving too much into tedium or over-realism.

### Proposed Food Groups:
- Vegetables
- Fruits
- Protein
- Grains

Naturally, in the case of Herbivores and Carnivores, the relevant cut-off food group should be made not required.
If one adequately manages their food groups, they get a bonus of some kind. If they are lacking in a certain group, they get a malus. The maluses should probably compound for each group you're lacking in

### OPTIONAL / OTHER IDEAS
The mod I got inspired from also has a Sugar category. As such, we could have a Sugar / Junkfood category that works similarly, where it's a non-essential food group that provides a malus if *over-consumed*. However, this would probably be taking the place (or at least justifying a rework of) the current Junkfood downsides.

Perhaps there should also be a penalty for over-eating in certain categories? The mod I was inspired by had an overeating penalty for all except vegetables, which makes some sense. Additionally, it actually has stages with bonuses, maluses **and** a neutral zone on either end of the 'target' zone. 

### Final Notes:
This would probably require some amount of JSON-ization of mutations or implementation of flags to properly disable the groups for the mutations that exclude a group. Then again, pretty much any good vitamin system would?

Feel free to do additional cogitating about this, I fully recognize the idea may not be quite fully baked yet.

**Comment by ghost** [comment]
I came to the conclusion on discord that no vitamin system can work without JSONizing it into mutations so mods and mainline mutations can all affect it. See below:

An example of why we shouldn't tie it to hardcode can be found in the UPS, my eternal nemesis:

https://cdn.discordapp.com/attachments/830916329053487124/1177741055035195412/image.png?ex=65739bd4&is=656126d4&hm=954ae3010407ae53e66d9fec8780f8418360380a237176a3c667a727a3ecb518&

https://cdn.discordapp.com/attachments/830916329053487124/1177741986040643664/image.png?ex=65739cb2&is=656127b2&hm=fe541e84b50c93cd2a9f71a84ec98345b2e43a4323c2ae156e469db4d7262ce2&

The UPS is tied to item ID so we're stuck with one and an advanced one. Now for vitamins: Any radical system that matters in gameplay will be forced on players and modders and without a system to update said balance via JSONized flags or fields to change which nutrients are required and in which amounts will mean carnivore or herbivore players with food restrictions become crippled. 

We can use a flag just to outright disable it similar to flags we already have. A field would allow us to heavily balance mutations, like requiring double fat consumption for ursine players if we want to do that.

https://cdn.discordapp.com/attachments/830916329053487124/1177742272780046356/image.png?ex=65739cf7&is=656127f7&hm=ee77a05572ec2c0c4ab80b4791cf96fe6ae7943846ab64cf78611294480fb5a6&

**Reply by RobbieNeko** [reply]
100% agreed. Any sort of real vitamin system (at least one meant to encourage variety) is going to need that due to the mutations that cut off available food groups / types.

**Reply by scarf005** [reply]
kind of tangential, but images can be rendered in markdown using

`![](the URL link of image to render)`

for example, this renders into

```
![](https://cdn.discordapp.com/attachments/830916329053487124/1177741986040643664/image.png?ex=65739cb2&is=656127b2&hm=fe541e84b50c93cd2a9f71a84ec98345b2e43a4323c2ae156e469db4d7262c)
```

this:

![](https://cdn.discordapp.com/attachments/830916329053487124/1177741986040643664/image.png?ex=65739cb2&is=656127b2&hm=fe541e84b50c93cd2a9f71a84ec98345b2e43a4323c2ae156e469db4d7262c)

**Comment by scarf005** [comment]
maybe we could have
- **major groups** which only affects health (e.g protein) 
- **subgroups** which only affect morale and allergies (e.g dairy, meat, plant-based)

it'll be similar to how weapon categories are modelled.

[original link](https://discord.com/channels/830879262763909202/830916329053487124/1177756526954426388)

**Reply by RobbieNeko** [reply]
Very sensible idea 👍
